### DISCUSSION TITLE
Cannot upload .patch files?

### DISCUSSION THREAD

**Discussion by tildearrow** [community/community#70193] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

# Description
It appears I am unable to attach a file with .patch extension, despite the error message mentioning that .patch is an acceptable extension for upload.

# Steps to Reproduce
Try attaching a .patch file (in Issues, Pull requests or Discussions).

# Video

https://github.com/community/community/assets/13842506/d6ec2cbc-6c3c-4415-8eec-e1c4a2a2c4cf

**Comment by duskmoss** [comment]
This is also happening to me

**Comment by kyrylo-sovailo** [comment]
Same is happening to me right now. KDE/Mozilla 130.0.1 (64-bit)

**Comment by thodnev** [comment]
same issue with the patch
```patch
diff --git a/README.md b/README.md
index e31d001..7493550 100644
--- a/README.md
+++ b/README.md
@@ -144,6 +144,8 @@ require("typescript-tools").setup {
     -- locale of all tsserver messages, supported locales you can find here:
     -- https://github.com/microsoft/TypeScript/blob/3c221fc086be52b19801f6e8d82596d04607ede6/src/compiler/utilitiesPublic.ts#L620
     tsserver_locale = "en",
+    -- Node executable configuration (for example, bun)
+    tsserver_node_executable = "bun",
     -- mirror of VSCode's `typescript.suggest.completeFunctionCalls`
     complete_function_calls = false,
     include_completions_with_insert_text = true,
diff --git a/lua/typescript-tools/config.lua b/lua/typescript-tools/config.lua
index 2825b1d..e30d3d7 100644
--- a/lua/typescript-tools/config.lua
+++ b/lua/typescript-tools/config.lua
@@ -9,6 +9,7 @@
 ---@field tsserver_file_preferences table|fun(filetype: string): table
 ---@field tsserver_max_memory number|"auto"
 ---@field tsserver_locale string
+---@field tsserver_node_executable string
 ---@field complete_function_calls boolean
 ---@field expose_as_code_action ("fix_all"| "add_missing_imports"| "remove_unused" | "remove_unused_imports")[]
 ---@field include_completions_with_insert_text boolean
@@ -125,6 +126,11 @@ function M.load_settings(settings)
       "string",
       true,
     },
+    ["settings.tsserver_node_executable"] = {
+      settings.tsserver_node_executable,
+      "string",
+      true,
+    },
     ["settings.complete_function_calls"] = { settings.complete_function_calls, "boolean", true },
     ["settings.expose_as_code_action"] = {
       settings.expose_as_code_action,
@@ -163,6 +169,10 @@ function M.load_settings(settings)
     __store.tsserver_file_preferences = {}
   end
 
+  if not settings.tsserver_node_executable then
+    __store.tsserver_node_executable = "node"
+  end
+
   if not M.tsserver_log_level[settings.tsserver_logs] then
     __store.tsserver_logs = M.tsserver_log_level.off
   end
diff --git a/lua/typescript-tools/process.lua b/lua/typescript-tools/process.lua
index a1dd48f..d10763d 100644
--- a/lua/typescript-tools/process.lua
+++ b/lua/typescript-tools/process.lua
@@ -123,7 +123,7 @@ local function parse_response(initial_chunk, on_response)
 end
 
 function Process:start()
-  local command = is_win and "cmd.exe" or "node"
+  local command = is_win and "cmd.exe" or plugin_config.tsserver_node_executable
 
   if type(plugin_config.tsserver_max_memory) == "number" then
     table.insert(self.args, 1, "--max-old-space-size=" .. plugin_config.tsserver_max_memory)
@@ -131,7 +131,7 @@ function Process:start()
 
   if is_win then
     table.insert(self.args, 1, "/c")
-    table.insert(self.args, 2, "node")
+    table.insert(self.args, 2, plugin_config.tsserver_node_executable)
   end
 
   local args = {

```

**Comment by justin025** [comment]
I get the same error, I checked the networking tab and it seems to be a NS_BINDING_ERROR on Firefox. On chromium however I get a 422 status code with the response attached below. Its most likely a server side issue, hopefully github will take a look at this
```json
{
    "errors": [
        {
            "resource": "RepositoryFile",
            "code": "custom",
            "field": "content_type",
            "message": "content_type is not included in the list"
        },
        {
            "resource": "RepositoryFile",
            "code": "custom",
            "field": "name",
            "message": "name has a file extension that does not match the content type"
        }
    ]
}

**Comment by 10senura** [comment]
diff --git a/README.md b/README.md
index e31d001..7493550 100644
--- a/README.md
+++ b/README.md
@@ -144,6 +144,8 @@ require("typescript-tools").setup {
     -- locale of all tsserver messages, supported locales you can find here:
     -- https://github.com/microsoft/TypeScript/blob/3c221fc086be52b19801f6e8d82596d04607ede6/src/compiler/utilitiesPublic.ts#L620
     tsserver_locale = "en",
+    -- Node executable configuration (for example, bun)
+    tsserver_node_executable = "bun",
     -- mirror of VSCode's `typescript.suggest.completeFunctionCalls`
     complete_function_calls = false,
     include_completions_with_insert_text = true,
diff --git a/lua/typescript-tools/config.lua b/lua/typescript-tools/config.lua
index 2825b1d..e30d3d7 100644
--- a/lua/typescript-tools/config.lua
+++ b/lua/typescript-tools/config.lua
@@ -9,6 +9,7 @@
 ---@field tsserver_file_preferences table|fun(filetype: string): table
 ---@field tsserver_max_memory number|"auto"
 ---@field tsserver_locale string
+---@field tsserver_node_executable string
 ---@field complete_function_calls boolean
 ---@field expose_as_code_action ("fix_all"| "add_missing_imports"| "remove_unused" | "remove_unused_imports")[]
 ---@field include_completions_with_insert_text boolean
@@ -125,6 +126,11 @@ function M.load_settings(settings)
       "string",
       true,
     },
+    ["settings.tsserver_node_executable"] = {
+      settings.tsserver_node_executable,
+      "string",
+      true,
+    },
     ["settings.complete_function_calls"] = { settings.complete_function_calls, "boolean", true },
     ["settings.expose_as_code_action"] = {
       settings.expose_as_code_action,
@@ -163,6 +169,10 @@ function M.load_settings(settings)
     __store.tsserver_file_preferences = {}
   end
 
+  if not settings.tsserver_node_executable then
+    __store.tsserver_node_executable = "node"
+  end
+
   if not M.tsserver_log_level[settings.tsserver_logs] then
     __store.tsserver_logs = M.tsserver_log_level.off
   end
diff --git a/lua/typescript-tools/process.lua b/lua/typescript-tools/process.lua
index a1dd48f..d10763d 100644
--- a/lua/typescript-tools/process.lua
+++ b/lua/typescript-tools/process.lua
@@ -123,7 +123,7 @@ local function parse_response(initial_chunk, on_response)
 end
 
 function Process:start()
-  local command = is_win and "cmd.exe" or "node"
+  local command = is_win and "cmd.exe" or plugin_config.tsserver_node_executable
 
   if type(plugin_config.tsserver_max_memory) == "number" then
     table.insert(self.args, 1, "--max-old-space-size=" .. plugin_config.tsserver_max_memory)
@@ -131,7 +131,7 @@ function Process:start()
 
   if is_win then
     table.insert(self.args, 1, "/c")
-    table.insert(self.args, 2, "node")
+    table.insert(self.args, 2, plugin_config.tsserver_node_executable)
   end
 
   local args = {
